### PR TITLE
Setting: Retain Stripe Data, place in correct spot settings array

### PR DESCRIPTION
**Pull request by psealock**
#### Changes proposed in this Pull Request:

Explicitly place the "Retain Stripe Data" setting at the end of the `personal_data_retention` section of settings instead of appending it as the second to last entry in the settings array. 

-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [x] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

#### Avoids this issue when Core adds additional settings

screen shot 2019-03-05 at 11 04 50 am

#### Test instruction

1. WooCommerce > Settings > Accounts & Privacy
2. The "Retain Stripe Data" should be the last entry in the "Personal data retention" section.

### CODE REVIEW

**Comment by mikejolley** [comment]
LGTM although there seems to be a since tag modified which I'm unsure is correct.

Will defer to @woocommerce/extendables  :)

File: includes/admin/class-wc-stripe-privacy.php [hunk 1]
```diff
@@ -44,7 +44,18 @@ public function account_settings( $settings ) {
 			),
 		);
 
-		array_splice( $settings, ( count( $settings ) - 1 ), 0, $insert_setting );
+		$index = null;
+
+		foreach( $settings as $key => $value) {
```
Comment (roykho) [inline includes/admin/class-wc-stripe-privacy.php]: Needs a space after `foreach`

File: woocommerce-gateway-stripe.php [hunk 2]
```diff
@@ -24,7 +24,7 @@
 /**
  * WooCommerce fallback notice.
  *
- * @since 4.1.2
+ * @since 4.1.15
```
Comment (roykho) [inline woocommerce-gateway-stripe.php]: What is this change for?

Comment (psealock) [reply woocommerce-gateway-stripe.php]: Oh right this is different than the current version number. I'll change this back.

**Comment by psealock** [comment]
Thanks for the feedback @roykho. I addressed the changes you mentioned.

**Pull request merged**